### DISCUSSION TITLE
enh: pip open-webui[min] installation

### DISCUSSION THREAD

**Discussion by AntonKrug** [open-webui/open-webui#7034] [Ideas]
# Feature Request

**Is your feature request related to a problem? Please describe.**
The docker has a option for the cases when you run ollama and models on a separate machine, however local pip install is installing the fullest set of nvidia and cuda drivers which i will not use. Is there a way to make a very minimalistic install just for the front end UI without the bulky rest?

**Describe the solution you'd like**
Cleaner separation of backend and frontend when installing through pip.

**Describe alternatives you've considered**
I was trying to install it as is, but it's really unecesary bulky, the torch library depends on ram, the nvidia cuda drivers depend on storage, and they will not even get used as the machine run diferent a VM where there is a GPU passed through, has most of the ram and most of the storage allocated. I would strongly prefer to have this separation even if the UI would end up having some features disabled. Forcing me make huge LXC container. Making multi GB LXC container feels to me almost abusing the concept of lightweight containers. At what point it's actually maybe better to make a dedicated VM instead of container.

**Additional context**
I do not want to use docker as I already have LXC support in my proxmox and so far everything can be done through LXC containers, installing docker for just one app feels to me overkill.

**Comment by mahenning** [comment]
I just wanted to say that the pip "package[extra]" brackets are for **adding** extra stuff, not removing something. One could put cuda stuff in e.g. "open-webui[cuda]", but [min] to remove torch from the base package is not possible to my knowledge. And because torch is a dependency for both Whisper and SentenceTransformers, I don't think it goes into some [extra] brackets soon.
